test/apps/iodemo/io_demo.cc

```diff
@@ -172,11 +173,23 @@ template<class BufferType, bool use_offcache = false> class ObjectPool {
     virtual BufferType *construct() = 0;
 
 private:
+    void fill_offcache_queue()
+    {
+        while (_offcache_queue.size() < _num_offcache) {
+            _offcache_queue.push(get_free());
+        }
+    }
+
     inline BufferType *get_free()
     {
         BufferType *item;
 
         if (_free_stack.empty()) {
+            // Fill the offcache queue on first use. Assume the free stack will
+            // also be empty on the first use.
+            if (_offcache_queue.size() < _num_offcache) {
+                fill_offcache_queue();
+            }
```

Comment:
maybe fill from ctor? then no need to have `_num_offcache`

Reply:
in ctor we still not have context for pre-registration. it would require introducing "init" function which is uglier IMO.